students/briggsm/session03/slicing.py

```diff
@@ -0,0 +1,44 @@
+# Slicing Lab
+
+def firstlast(x):
+    '''First and last items exchanged'''
+    slice = x[-1] + x[0]
+    return slice
+
+def picketfence(x):
+    '''Every other item removed'''
+    picket = ""
+    for i in range(len(x)):
+        if i % 2 == 0:
+            picket += x[i]
+    return picket
+
+def checkfour(x):
+    '''first and last 4 items removed, and every other item in between'''
+    check = x[3:-4]
+    return check
+
+def reverseit(x):
+    '''Elements reversed (just with slicing)'''
+    max = len(x)
+    revstring = ""
+    for i in range(max):
+        revstring += x[(max-1)-i]
+    return revstring
+
+def thirdman(x):
+    '''Middle third, then last third, then the first third in the new order'''
+    max = len(x)
+    third = int(max/3)
+    sone = x[0:third]
+    stwo = x[(third):(third*2)]
+    sthree = x[(third*2):]
+    rs = stwo + sthree + sone
+    return rs
```

Comment:
Your notation makes it obvious what you are doing, but you could combine lines 33, 34, 35, 36:
```
rs = x[third:third*2 + x[third*2:] + x[0:third]
```
Note that the distance from the front and back are inverted, so 2/3 distance from the front is equal to 1/3 distance from the end, so you can even do:
```
rs = x[third:-third] + x[-third:] + x[:third]
```
It is always a pull between making it pretty and concise and making it readable, but you will get better at reading/writing compact code as you practice.